docs: add DataHub Cloud v2.1.0 release notes (#18500)
datahub-project/datahub: docs/managed-datahub/release-notes/v_2_1_0.md

- **Amazon S3** — catalog object-store folders without ingesting the underlying files via `emit_folders_only`.
- **Kafka Connect** — Iceberg Sink Connector and Snowflake Streaming Sink Connector support.
- **Dataplex** — ingestion source UI forms for the V2 experience.

**Ingestion infrastructure:**

- **Async emit by default for orchestration plugins** — Airflow, Dagster, Prefect, and Great Expectations default to async emit to reduce load at high volume (see Breaking Changes).
- **Column-level classifier removed** — the built-in `DataHubClassifier` and its dependency are gone; the classification framework remains (see Breaking Changes).
- **Workunit processor API** — helper functions replaced by `WorkunitProcessor` classes (see Breaking Changes).
- **mTLS client authentication** — the CLI and SDK can present a client certificate on every outbound HTTPS path via `DATAHUB_CLIENT_CERT_PATH` (and optionally `DATAHUB_CLIENT_KEY_PATH`). Fully backwards compatible.
- **Environment-based OAuth** — `DATAHUB_AUTH_TYPE` selects OAuth for the CLI, default sink, and env-resolved emitters, backed by reusable OAuth token providers for the Python clients.

> - **Opt-in Kafka default sink** for managed ingestion, and a `DATAHUB_SQLGLOT_DISABLE_C` escape hatch to force pure-Python SQLGlot.

Reviewer: This should be split into two different lines.

Author: Feel free to push updates to the documentation as needed.

- **Agent, Service, and Repository catalog entities** — catalog AI agents, the skills they adopt and tools they invoke, the services those tools belong to (REST APIs, MCP servers), and the source-code repositories behind them as first-class entities that participate in lineage, with auto-registration available for LangChain and Google ADK agents. See the [Agent Registry](https://docs.datahub.com/docs/features/feature-guides/agent-registry) ([API tutorial](https://docs.datahub.com/docs/api/tutorials/agent-registry)) and [Service Catalog](https://docs.datahub.com/docs/features/feature-guides/service-catalog) guides.
- **Bi-Directional Context Document Sync for GitHub** - import context documents from one or more GitHub repositories, and sync changes made on DataHub back as GitHub PRs. Read more [here](https://docs.datahub.com/docs/features/feature-guides/context/context-documents#importing-documents).

> - **Expanded internationalization** — multi-language support is now on by default, and DataHub follows the browser locale where a translation exists.

Reviewer 2: The env var is on by default, but our DataHub Cloud helm charts still overwrite it to be disabled. Are we okay enabling it by default for Cloud customers or should we clarify here that customers can request enabling the feature from their DataHub contacts?
CC: @benjiaming

Reviewer 2: May want to mention the new locales explicitly: FR, IT, SV, NB, and HU

Author: cc: @manuelawei8

---
description: "DataHub Cloud v2.1.0 release notes — Metrics and Semantic Models, scoped MCP servers, Agent/Service/Repository catalog entities, Ask DataHub GA with human-in-the-loop tool approval and multi-provider LLMs, a Logical Models UI, expanded internationalization, and a wave of new and improved ingestion sources."
---

# v2.1.0

## Release Changelog

### v2.1.0

#### Release Availability Date

TBD

#### Recommended Versions

- **CLI/SDK**: 1.6.0.16

> - **Remote Executor**: v2.1.0-cloud, v2.0.4-cloud, v2.0.3-cloud, v2.0.2-cloud, v2.0.1-cloud, v1.1.4-cloud (Only with Kafka Queues, Not supported with SQS queues)

Author: @skrydal @rajatoss Please confirm the RE support versions and update if any changes need to be made.

Reviewer 3: > v1.1.4-cloud (Only with Kafka Queues, Not supported with SQS queues)

v1.1.4-cloud works with sqs if a customer has only sqs channel, it doesn't work with sqs if you have some kafka channels and some sqs channel. Kafka with this version is only rolled out to yahoo, for them also Piotr has updated to 2.0.0.
In the last release doc we mentioned that. 
`You deploy Kafka-enabled executors (version `≥ 2.0.0`, `channel: KAFKA`) into the pool. They join but stay in `Waiting` state - they do not receive work yet.` Hence KAFKA is after 2.0.0.

On deploy-test these fails with sqs on v1.1.4-cloud because we test in a setup which has both pools.

CC: @skrydal

> - **Remote Executor**: v2.1.0-cloud, v2.0.4-cloud, v2.0.3-cloud, v2.0.2-cloud, v2.0.1-cloud, v1.1.4-cloud (Only with Kafka Queues, Not supported with SQS queues)

Reviewer 3: ```suggestion
- **Remote Executor**: v2.1.0-cloud, v2.0.4-cloud, v2.0.3-cloud, v2.0.2-cloud, v2.0.1-cloud, v1.1.4-cloud
```

- **Power BI and Mode column-level lineage preserves original casing.** These connectors no longer lowercase upstream column names, so previously-missing column-level edges will render after re-ingestion. If you set `convert_column_urns_to_lowercase: true` on an upstream SQL source as a workaround, remove it so both sides preserve casing.
- **BigQuery table statistics source changed.** Row count, size, and last-altered now come from `INFORMATION_SCHEMA.PARTITIONS` instead of the undocumented `__TABLES__`. `PARTITIONS` covers base tables only, so empty tables, external tables, views, and snapshots lose their `lastModified` timestamp (and snapshots lose row/size custom properties). Set `use_legacy_table_stats: true` to restore the previous behavior.

> - **MongoDB can be ingested as AWS DocumentDB.** A new opt-in `platform: documentdb` (with `hostingEnvironment: AWS_DOCUMENTDB`) emits entities under a new `documentdb` data platform. The default is unchanged; enabling it changes dataset and container URNs, and stale `mongodb` entities must be cleaned up.

Author: Is this MongoDb update really a breaking change?

- **Snowplow enrichment DataJob URNs are now name-based.** The connector moved to the supported enrichments endpoint, which no longer returns per-enrichment UUIDs, so enrichment DataJob URNs derive from the enrichment name. Old UUID-based DataJobs are re-created; clean up stale entities.

> - **Teradata database-container URN casing.** When `convert_urns_to_lowercase: true` (non-default), the connector now also lower-cases database-container URNs, not just datasets, so containers and their datasets match. Deployments on the default (`false`) are unaffected.

Author: This is not a breaking change I believe if the default is false.

> - **Ingestion framework: workunit processor API.** Workunit-processor helper functions have been removed in favor of `WorkunitProcessor` classes, and several processors were renamed. Custom code that called the helper functions directly must migrate to the processor-class API.

Author: @skrydal does this need to be in the public docs?

Reviewer: I didn't put it here. In general those are put in the CLI release notes (publicly though). It is a big change though, it is good if the knowledge is spread.
CC: @sgomezvillamor

Reviewer 4: Yeah, this breaking change needs to be public, and it was already announced in the corresponding `acryl-datahub` release notes.

Whether ingestion notes need to be included in the saas release notes, that's a separate discussion, raised it here https://github.com/datahub-project/datahub/pull/18500#discussion_r3664542587

- **Dataset semantic search is no longer implicit.** Enabling semantic search with the default entity set no longer also bridges datasets; set `ELASTICSEARCH_SEMANTIC_SEARCH_ENTITIES=document,dataset` on GMS and the system-update job to keep dataset semantic search after upgrade.
- **Logical parent links require Edit Entity on both datasets.** Linking a physical dataset to a logical parent now requires Edit Entity on both the child and the proposed parent. Clearing a parent requires Edit Entity on the child only. Grant the privilege or update policies accordingly.

> - **GMS usage-metric semantics (opt-in).** With `USAGE_AGGREGATION_ENABLED=true`, `datahub_request_count` is emitted from the in-memory aggregation flush to Micrometer/Prometheus (not per-request) and the actor tag becomes `actor_class`. New byte and active-identity metrics are added. Aggregation is off by default; legacy JMX metrics are unaffected.

Author: Not a breaking change I think. @david-leifker

#### Product

- **Metrics and Semantic Models** — new Metric and SemanticModel entity pages, a metrics home page, a metrics sidebar, and autocomplete with infinite scroll, so metrics and the semantic models that define them are browsable and searchable like any other asset.
- **Logical Models UI** — a "Create Logical Model" action in the search bar opens a form to define a model's name, logical platform, and columns. On a logical model's profile, a "Physical Children" sidebar links physical datasets and maps their columns to the logical parent, and the schema tab supports adding, renaming, retyping, and deleting columns. Creating and editing a logical model requires the new "Create Logical Models" platform privilege (granted to Admin by default).
- **Lineage graph controls** — added a redraw control, an empty-graph nudge, and a modernized control layout for the lineage visualization.
- **Domains** — domains can now be attached to Glossary Nodes, and the Manage Domains sidebar reaches parity with the documents experience.
- **Document search scoped to the active View** — document search now respects the active View, so results stay within the working set the user has selected.

> - **Internationalization** — multi-language support defaults on and follows the browser locale; French, Italian, Norwegian, Swedish, and Hungarian locales are available (several still Beta). Users can always pick their language under Settings → Preferences.

Reviewer 5: ```suggestion
- **Internationalization** — multi-language support defaults on and follows the browser locale; German (GA), Spanish, Brazilian Portuguese, French, Italian, Norwegian, Swedish, and Hungarian locales are available (Private Beta). Users can always pick their language under Settings → Preferences.
```

- **Workunit processor API** — helper functions replaced by `WorkunitProcessor` classes (see Breaking Changes).
- **mTLS client authentication** — the CLI and SDK can present a client certificate on every outbound HTTPS path via `DATAHUB_CLIENT_CERT_PATH` (and optionally `DATAHUB_CLIENT_KEY_PATH`). Fully backwards compatible.
- **Environment-based OAuth** — `DATAHUB_AUTH_TYPE` selects OAuth for the CLI, default sink, and env-resolved emitters, backed by reusable OAuth token providers for the Python clients.
- **Opt-in Kafka default sink** for managed ingestion, and a `DATAHUB_SQLGLOT_DISABLE_C` escape hatch to force pure-Python SQLGlot.
- **Temp-table usage stats fixed** — per-query popularity statistics for queries that load through temporary/staging tables (the common ELT/dbt pattern) now land on the composite Query that lineage references, across every connector on the shared SQL parsing aggregator.

**Executor:**

- **Ingestion-log garbage collector** — opt-in in-process cleanup of per-execution log directories older than a retention window, with a size-cap safety net (`DATAHUB_EXECUTOR_LOG_GC_ENABLED`, default `false`).

> - **Stored-procedure CALL statements** — custom SQL assertions can optionally allow a `CALL` statement (`DATAHUB_EXECUTOR_ALLOW_CALL_STATEMENTS`, default `false`).

Reviewer 6: @gabe-lyons @treff7es @sgomezvillamor @naeemdh @skrydal please take a look at this section - review for accuracy & add in any other highlights worth calling out

Reviewer 4: That's a bit odd — ingestion version lifecycle is decoupled from the server version. We publish ingestion much more frequently, and the server supports a wide range of CLI versions.
Do we usually include ingestion updates in the SaaS server release notes?

That said, the connector improvements and ingestion infra changes look correct (or at least familiar) to me — just not sure which CLI version(s) were considered here.

Author: This was generated using the the skill @maggiehays provided [https://github.com/acryldata/datahub-skills/tree/main/skills/cloud-release-notes.](<https://github.com/acryldata/datahub-skills/tree/main/skills/cloud-release-notes.>) <br>Not sure which CLI version this referred to.

Reviewer 7: Can we just point to the DataHub CLI release changelog? -> https://github.com/acryldata/datahub/releases/tag/v1.6.0.16

Reviewer 7: This is it to be more precise because the previous default was 1.6.0.4

Version | Release Date | Link
-- | -- | --
v1.6.0.5 | 2026-06-19 | https://github.com/acryldata/datahub/releases/tag/v1.6.0.5
v1.6.0.6 | 2026-06-25 | https://github.com/acryldata/datahub/releases/tag/v1.6.0.6
v1.6.0.7 | 2026-06-26 | https://github.com/acryldata/datahub/releases/tag/v1.6.0.7
v1.6.0.8 | 2026-06-28 | https://github.com/acryldata/datahub/releases/tag/v1.6.0.8
v1.6.0.9 | 2026-06-30 | https://github.com/acryldata/datahub/releases/tag/v1.6.0.9
v1.6.0.10 | 2026-07-05 | https://github.com/acryldata/datahub/releases/tag/v1.6.0.10
v1.6.0.11 | 2026-07-09 | https://github.com/acryldata/datahub/releases/tag/v1.6.0.11
v1.6.0.12 | 2026-07-09 | https://github.com/acryldata/datahub/releases/tag/v1.6.0.12
v1.6.0.13 | 2026-07-09 | https://github.com/acryldata/datahub/releases/tag/v1.6.0.13
v1.6.0.14 | 2026-07-14 | https://github.com/acryldata/datahub/releases/tag/v1.6.0.14
v1.6.0.15 | 2026-07-16 | https://github.com/acryldata/datahub/releases/tag/v1.6.0.15
v1.6.0.16 | 2026-07-27 | https://github.com/acryldata/datahub/releases/tag/v1.6.0.16

- **GMS rate-limit configuration renamed.** `defaultRetryAfterSeconds` / `RATE_LIMITS_DEFAULT_RETRY_AFTER` are renamed to `minRetryAfterSeconds` / `RATE_LIMITS_MIN_RETRY_AFTER` (now a minimum floor), and a retry-after jitter percentage is added. An external rate-limit policy file is now loaded by Spring: `RATE_LIMITS_CONFIG_FILE_ENABLED` and `RATE_LIMITS_CONFIG_JSON` are removed, `RATE_LIMITS_CONFIG_FILE` must be a scheme-qualified URI (for example `file:/etc/datahub/rate-limits.yaml`), and a mounted policy file must be rooted under `datahub.gms.rateLimits:`. Migrate these before upgrading if you set them.
- **Semantic-anchor ingestion: `output_dir` removed.** The `semantic-anchor` source no longer accepts `output_dir`; artifacts are written to `INGESTION_ARTIFACT_DIR`, which the executor sets automatically. Remove `output_dir` from recipes.
- **Executor coordinator env flags.** `DATAHUB_EXECUTOR_MONITORS_ENABLED` and `DATAHUB_EXECUTOR_TASKS_ENABLED` are now hard opt-outs that skip subsystem wiring and heavy imports, not fetcher-only toggles. Use `DATAHUB_EXECUTOR_INGESTION_PIPELINE_ENABLED` to disable the Kafka / actions pipeline.

> - **Smart-assertion inference selector split.** `DATAHUB_USE_OBSERVE_MODELS=true` no longer implies the V2 trainer. Any executor relying on it for V2 must also set `DATAHUB_USE_INFERENCE_V2=true`, or it falls back to the V1 trainer after upgrade.

Reviewer 8: ```suggestion
```

Author: @AdrianMachado Do we need to update the env variables in https://github.com/acryldata/datahub-executor-helm for people running Assertions using Remote Executor?

Reviewer 8: Good question - let me take a look

- **Multi-provider LLM support.** A single `DATAHUB_LLM_PROVIDER` setting selects the provider (Bedrock, Anthropic, OpenAI, or Vertex AI) whose recommended defaults apply across every LLM use case. When unset, the provider is auto-detected from the environment (Vertex AI on GCP, otherwise Bedrock). Bare `claude-*` model IDs route to the Anthropic API directly, and the remaining SQL-context, smart-search, and constraint-parsing call sites now honor any provider prefix.
- **Managed model upgrades.** Managed (non-BYOK) Bedrock defaults for the Sonnet-tier use cases move to the latest Claude Sonnet models. Deployments that pinned a specific model keep it.
- **Answers finalize as plain text.** The chat agent no longer relies on an internal tool to deliver its final answer — it ends its turn with the answer, which is more reliable across models. Entity links are rewritten to canonical frontend URLs in stored history, and follow-up suggestions are generated by a dedicated call (`CHAT_SUGGESTIONS_ENABLED`, default on) delivered on their own stream event so they never delay the reply.

> - **Slack experience.** The connect prompt now leads with what DataHub does and offers a one-click "I've connected" button that replays the user's original question. Slack and Microsoft Teams connection state is now read from dedicated GraphQL status queries (a single source of truth) rather than decoding the credential blob, fixing settings pages that could show a working integration as disconnected.

Reviewer 8: ```suggestion
- **Slack experience.** The connect prompt now leads with what DataHub does and offers a one-click "I've connected" button that replays the user's original question. 
```

> - **Custom agents billed separately from Ask DataHub.** Custom agents now report distinct usage telemetry and no longer consume the Ask DataHub message-credit pool. Core Ask DataHub modes, the Ingestion Troubleshooter, and the Data Observability agent remain classified as Ask DataHub usage. No customer-facing behavior change — this only affects usage categorization.

Reviewer 8: ```suggestion
```

- **Context Hub clustering.** Metadata-observation clustering can propose grouped context, route proposals to context-document owners, and surface them in the inbox with an accept/reject flow.

#### Observe

- **Exclusion windows record `INIT`.** When a smart (inferred) assertion is evaluated while "now" falls inside one of its monitor's exclusion windows, the executor samples the metric but records the run as `INIT` rather than computing pass/fail, so a planned freeze, migration, or backfill no longer produces spurious anomalies. The window name is surfaced on the run and in the assertion drawer.
- **Smart-assertion inference V2.** Smart-assertion training can route to a V2 trainer pipeline, selected by the new `DATAHUB_USE_INFERENCE_V2` flag (default `false`); this is now decoupled from `DATAHUB_USE_OBSERVE_MODELS`.
- **Delta-space bounds gating.** The V1 trainer only explores and persists differenced (delta-space) prediction bounds when `DATAHUB_EXECUTOR_ENABLE_DELTA_BOUNDS=true` (default `false`), protecting mixed-fleet rollouts from misreading delta bounds as absolute.

> - **Lower-footprint executor builds.** With `DATAHUB_USE_OBSERVE_MODELS=false`, the V1 trainer no longer imports the observe-models package (PyTorch/Darts) at module load, so slim and low-RAM executor builds no longer risk an import-time crash loop.

Reviewer 8: ```suggestion
- **Redesigned notification emails.** Assertion notification emails have a refreshed layout and more useful details about the assertion and the failure that's easier to scan. The email body also now includes an `alert id` that can be used for deduplicating alerts in your inbox or through external system integrations.
- **Subscriptions follow their asset's lifecycle.** When an asset you're subscribed to is soft-deleted, its subscriptions are deactivated automatically, so you stop receiving alerts for an asset that no longer exists; restoring the asset brings them back. A background reconciliation keeps subscriptions consistent with their target's status even if an update is missed.
- 
- **Snowflake Column Casing** Adds support for case-sensitive column identifiers in Snowflake assertion queries through an opt-in configuration flag: `DATAHUB_EXECUTOR_SNOWFLAKE_QUOTE_COLUMNS`. Turn this flag on if your column names are case sensitive in Snowflake.
```

- **Spring Security and commons-configuration2**: bumped to address Java CVEs.
- **starlette**: bumped to 1.3.1 to address CVE-2026-48710, CVE-2026-48817, CVE-2026-48818, and CVE-2026-54283.
- **redshift-connector**: minimum bumped to 7.45.5-equivalent floor (`>=2.1.14`) to address CVE-2026-8838.
- **joserfc and soupsieve**: CVE floors raised for the executor and integrations service.

#### Bug Fixes

- Fixed keyed-array patch rebase dropping the map key when a patch created a new element through a deeper path (for example adding a field-level tag or term to a schema field with no existing editable metadata), which previously failed with an `HTTP 422 - fieldPath is required`.
- Raised the JSON property-name length limit on metadata deserialization so deeply-nested struct field paths (for example dbt column-level lineage carried in patches) no longer fail with a name-length exception.
- Fixed oversized `queryProperties` aspects failing ingestion runs by measuring serialized size correctly; oversized statements are now truncated with a warning instead of failing the run.
- Improved Ask DataHub and chat reliability, including prompt-cache friendliness and delegation-routing latency.

> - Improved server-side performance for glossary root loading, entity-registry schema scans, and assertion-assignment GraphQL batching.

Reviewer 8: ```suggestion
- Improved server-side performance for glossary root loading, entity-registry schema scans, and assertion-assignment GraphQL batching.
- Fixed Slack and Microsoft Teams settings pages that could show a connected integration as disconnected.
```
